### DISCUSSION TITLE
Per route SPA in statically generated site

### DISCUSSION THREAD

**Discussion by marekmis** [nuxt/nuxt#17234] [Questions]
I know there are previous requests and discussions going on around [Per route SSR/SPA](https://github.com/nuxt/framework/discussions/4665), but it would be really cool to have an option to define SPA handling of selective routes when the entire site is statically generated with `nuxi generate`

So for example we'd have a statically generated page listing cars _'/cars'_  and a dynamic page _'/cars/car1'_, the browser would hit the _'/cars'_ and nuxt redirect to _'/cars/car1'_. 

Is there a way to achieve this with rc4?

**Comment by mirumirumi** [comment]
Hi!
I did exactly that the other day, implementing only the `?page=2` pagination part with CSR(=SPA).

Monitor the root object and render (+data fetch) on the client side only when necessary:
https://github.com/mirumirumi/mirumi-tech-frontend/blob/main/src/pages/index.vue#L40-L83

Yes, I agree that it would be useful to be able to configure by configuration, but unlike SSR, ISR, etc., I have my doubts about the user experience of suddenly turning the entire verge into a CSR.

Rather, it is better to be able to provide dynamic content within a single page, which matches the use case of SPA and can be controlled by query parameters as described above.

This is just my opinion.
Have a nice day! :)
